sqlalchemy_continuum/manager.py

```diff
@@ -431,6 +431,21 @@ def track_cloned_connections(self, c, opt):
                 if not connection.closed and connection.connection is c.connection:  # ConnectionFairy is the same - this is a clone
                     self.units_of_work[c] = uow
 
+    def get_python_processed_parameter(self, parameter, column, dialect):
+        param_transformer = column.type.dialect_impl(dialect)
+        process = param_transformer.result_processor(dialect, column.type)
+        if process is not None:
+            # https://github.com/sqlalchemy/sqlalchemy/blob/231262d4782fc4d1a5d5fdbf985a06e04270e878/lib/sqlalchemy/sql/type_api.py#L452
+            return process(parameter)
+        return parameter
+
+    def process_parameters(self, table_name, parameters, dialect):
+        table = self.metadata.tables[table_name]
+        return tuple(
+            self.get_python_processed_parameter(parameter=parameters[idx], column=column, dialect=dialect) 
+            for idx, column in enumerate(table.columns)
```

Comment:
I am not sure I get this.
You are doing a `enumerate(table.columns)` - so lets say I have 10 columns in my table.
Not you do `parameters[idx]` - You are assuming every parameter corresponds to each column ?

There are some asusmptions you seem to be making:
1. This is only valid for INSERT statements
2. Are we sure that ALL columns will be present ? What if there is a `server_default` or nullabe=True which makes the column null by default ?

Reply:
In fact, now that I look at it more - isn't `parameters` a dictionary ?
Shouldn't we be doing: `parameters[column.key]` ?

Reply:
yes, i think for defaults it does return some value from SQLA itself, however the issue you are pointing to i have raised same in thread for #283 for delete query perspective and i think it does exist currently in master branch also, a seperate thread is on ig for refactoring `positional_args_to_dict`

` isn't parameters a dictionary ?` - No it's a tuple `positional_args_to_dict` converts this tuple to dict